File: src/main/java/com/github/codingvampyre/drachenfest/controllers/AccountController.java
package com.github.codingvampyre.drachenfest.controllers;

import com.github.codingvampyre.drachenfest.domain.Account;
import com.github.codingvampyre.drachenfest.dtos.ParticipantCreateRequestDto;
import com.github.codingvampyre.drachenfest.services.participant.ParticipantService;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.List;

@RestController
public class AccountController {

    private final ParticipantService participantService;

    AccountController(ParticipantService participantService) {
        this.participantService = participantService;
    }

    @GetMapping("/participants")
    List<Account> list() {
        return this.participantService.findAll();
    }

    @GetMapping("/participants/{email}")
    Account describeByUuid(@PathVariable String email) {
        return this.participantService.findByemail(email);
    }

    @PostMapping("/participants")
    Account create(@RequestBody @Valid ParticipantCreateRequestDto participantCreateRequestDto) {
        return this.participantService.create(participantCreateRequestDto);
    }

}

File: src/main/java/com/github/codingvampyre/drachenfest/controllers/AccountController.kt
package com.github.codingvampyre.drachenfest.controllers

import com.github.codingvampyre.drachenfest.domain.Account
import com.github.codingvampyre.drachenfest.dtos.ParticipantCreateRequestDto
import com.github.codingvampyre.drachenfest.services.participant.ParticipantService
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
class AccountController internal constructor(private val participantService: ParticipantService) {
	@GetMapping("/participants")
	fun list(): List<Account> {
		return participantService.findAll()
	}

	@GetMapping("/participants/{email}")
	fun describeByUuid(@PathVariable email: String): Account? {
		return participantService.findByemail(email)
	}

	@PostMapping("/participants")
	fun create(@RequestBody participantCreateRequestDto: @Valid ParticipantCreateRequestDto): Account {
		return participantService.create(participantCreateRequestDto)
	}
}